routers/web/repo/pull.go

```diff
@@ -319,8 +319,37 @@ func PrepareMergedViewPullInfo(ctx *context.Context, issue *models.Issue) *git.C
 	setMergeTarget(ctx, pull)
 	ctx.Data["HasMerged"] = true
 
+	var baseCommit string
+	// Some migrated PR won't have any Base SHA and lose history, try to get one
+	if pull.MergeBase == "" {
+		// If there is a patch file, and it is readable, grab info
+		commitSHA, err := ctx.Repo.GitRepo.ReadPatchCommit(pull.Index)
+		if err != nil {
+			log.Trace("Pull request without history, %v", err)
+		} else {
+			// Get immediate parent of the first commit in the patch, grab history back
+			parentCommit, err := git.NewCommand("rev-list", "-1", "--skip=1", commitSHA).RunInDir(ctx.Repo.GitRepo.Path)
```

Comment:
NewCommandContext instead?

Reply:
I tried with NewCommandContext, but I don't understand why it is better ? 
I just had to change logic a little to get it to work with NewCommandContext, and it works the same... But I can commit it if you prefer !

Reply:
It works without a context, the only problem is the `command` can not be stopped if the request has been cancelled.

With a context, all related go-routines can be cancelled as soon as the context is cancelled.

In future, Gitea functions will use context as much as possible.

Reply:
Thanks for this explanation, I used the Context function ! 👍